File: sdekapi/warehouseservice/src/main/java/com/example/warehouseservice/warehouseservice/repository/WarehouseRepositoryImpl.java
package com.example.warehouseservice.warehouseservice.repository;

import com.eventstore.dbclient.EventData;
import com.eventstore.dbclient.EventStoreDBClient;
import com.example.warehouseservice.warehouseservice.model.Warehouse;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.scheduling.annotation.Async;
import org.springframework.stereotype.Service;
import java.util.concurrent.CompletableFuture;
import java.util.concurrent.ExecutionException;

@Service
public class WarehouseRepositoryImpl implements WarehouseRepository {
    EventStoreDBClient client;

    @Autowired
    public WarehouseRepositoryImpl(EventStoreDBClient client) {
        this.client = client;
    }

    @Async
    @Override
    public CompletableFuture<Boolean> addWarehouse(Warehouse warehouse) throws ExecutionException, InterruptedException {
        EventData data = EventData.builderAsJson("warehouse_add", warehouse).build();
        client.appendToStream("warehouse", data).get();
        return CompletableFuture.completedFuture(true);
    }

    @Async
    @Override
    public CompletableFuture<Boolean> updateWarehouse(Warehouse warehouse) throws ExecutionException, InterruptedException {
        EventData data = EventData.builderAsJson("warehouse_update", warehouse).build();
        client.appendToStream("warehouse", data).get();
        return CompletableFuture.completedFuture(true);
    }

    @Async
    @Override
    public CompletableFuture<Boolean> deleteWarehouse(Warehouse warehouse) throws ExecutionException, InterruptedException {
        EventData data = EventData.builderAsJson("warehouse_delete", warehouse).build();
        client.appendToStream("warehouse", data).get();
        return CompletableFuture.completedFuture(true);
    }
}

File: sdekflutterclient/lib/data/repositoryimpl/warehouse_repository_impl.dart
import 'package:clientapp/data/repository/warehouse_data_repository.dart';
import 'package:clientapp/domain/model/Warehouse.dart';
import 'package:clientapp/domain/repository/warehouse_repository.dart';

class WarehouseRepositoryImpl implements WarehouseRepository {
  final WarehouseDataRepository repository;

  WarehouseRepositoryImpl({
    required this.repository
  });

  @override
  Future<Warehouse> getWarehouseById(int id) async {
    return await repository.getWarehouseById(id);
  }

  @override
  Future<List<Warehouse>> getWarehouses() async {
    return repository.getWarehouses();
  }

  @override
  Future<void> addWarehouse(Warehouse warehouse) async {
    await repository.addWarehouse(warehouse);
  }

  @override
  Future<void> deleteWarehouse(Warehouse warehouse) async {
    await repository.deleteWarehouse(warehouse);
  }

  @override
  Future<void> updateWarehouse(Warehouse warehouse) async {
    await repository.updateWarehouse(warehouse);
  }
}